tests/storage/glusterfs/conftest.py

```diff
@@ -0,0 +1,128 @@
+import pytest
+
+def _setup_host_with_glusterfs(host):
+    assert not host.file_exists('/usr/sbin/glusterd'), \
+        "glusterfs-server must not be installed on the host at the beginning of the tests"
+    host.yum_install(['glusterfs-server'], save_state=True)
+    host.ssh(['systemctl', 'enable', '--now', 'glusterd.service'])
+
+    for h in host.pool.hosts:
+        hostname_or_ip = h.hostname_or_ip
+        if hostname_or_ip != host.hostname_or_ip:
+            host.ssh(['iptables', '-I', 'INPUT', '-p', 'all', '-s', hostname_or_ip, '-j', 'ACCEPT'])
+
+    # Make rules reboot-persistent
+    for service in ['iptables', 'ip6tables']:
+        host.ssh(['cp', '/etc/sysconfig/%s' % service, '/etc/sysconfig/%s.orig' % service])
+        host.ssh(['service', service, 'save'])
+
+def _teardown_host_with_glusterfs(host):
+    try:
+        host.ssh(['systemctl', 'disable', '--now', 'glusterd.service'])
+    except Exception:
+        pass
+    host.yum_restore_saved_state()
+
+    for h in host.pool.hosts:
+        hostname_or_ip = h.hostname_or_ip
+        if hostname_or_ip != host.hostname_or_ip:
+            host.ssh(['iptables', '-D', 'INPUT', '-p', 'all', '-s', hostname_or_ip, '-j', 'ACCEPT'])
+
+    for service in ['iptables', 'ip6tables']:
+        host.ssh(['mv', '/etc/sysconfig/%s.orig' % service, '/etc/sysconfig/%s' % service])
+
+@pytest.fixture(scope='session')
+def pool_with_glusterfs(host):
+    for h in host.pool.hosts:
+        _setup_host_with_glusterfs(h)
+
+    yield
+    for h in host.pool.hosts:
+        _teardown_host_with_glusterfs(h)
```

Comment:
If one host fails, we should collect the error, still do it on the others, and then raise with all errors grouped